packages/core/src/builtin-tasks/contract.ts

```diff
@@ -0,0 +1,209 @@
+import { paramTypes } from '../params'
+import { getUserConfigPath } from '../project-structure'
+
+task('contract').setAction(async () => {
+  execSync('kuai contract --help', { stdio: 'inherit' })
+})
+
+subtask('contract:get-workspace').setAction(async (_, { config }) => {
+  if (config.contract?.workspace) {
+    return config.contract?.workspace
+  }
+
+  const userConfigPath = getUserConfigPath()
+  if (!userConfigPath) {
+    throw new Error('Please run in kuai project file')
+  }
+
+  return path.join(path.dirname(userConfigPath), 'contract')
+})
+
+subtask('contract:set-environment').setAction(async () => {
+  // todo: download ckb-cli & capsule etc...
+})
+
+interface ImportArgs {
+  privateKey: string
+}
+
+subtask('contract:import-private-key')
+  .addParam('privateKey', '', '', paramTypes.string, false)
+  .setAction(async ({ privateKey }: ImportArgs) => {
+    const lockArg = key.privateKeyToBlake160(privateKey)
+
+    const res = execSync('ckb-cli account list --output-format json').toString()
+    const accountList: Array<{ lock_arg: string }> = JSON.parse(res)
+
+    if (accountList.some((account) => account.lock_arg === lockArg)) {
+      return
+    }
+
+    fs.writeFileSync('./.tmp_pk', privateKey)
+
+    console.log('add the user to ckb-cli, enter a password for the signature please')
+    execSync('ckb-cli account import --privkey-path ./.tmp_pk', { stdio: 'inherit' })
+
+    fs.rmSync('./.tmp_pk')
+  })
+
+interface BuildArgs {
+  name?: string
+  release?: boolean
+}
+
+subtask('contract:build')
+  .addParam('name', 'contract name', '', paramTypes.string, true)
+  .addParam('release', 'Build contracts in release mode', false, paramTypes.boolean, true)
+  .setAction(async ({ name, release }: BuildArgs, { run }) => {
+    const workspace = await run('contract:get-workspace')
+    execSync(`cd ${workspace} && capsule build${name ? ` --name ${name}` : ''}${release ? ' --release' : ''}`, {
+      stdio: 'inherit',
+    })
+  })
+
+interface NewArgs {
+  name: string
+  template: string
+}
+
+subtask('contract:new')
+  .addParam('name', 'The name of new contract')
+  .addParam(
+    'template',
+    'language template  [default: rust]  [possible values: rust, c, c-sharedlib]',
+    'rust',
+    paramTypes.string,
+    true,
+  )
+  .setAction(async ({ name, template }: NewArgs, { run }) => {
+    const workspace = await run('contract:get-workspace')
+    console.log('workspace', workspace)
+    execSync(`cd ${workspace} && capsule new-contract ${name} --template ${template}`, { stdio: 'inherit' })
+  })
+
+interface DeployArgs {
+  name?: string
+  chain?: string
+  isMainnet?: boolean
+  fee: number
+  env: string
+  migrate: string
+}
+
+subtask('contract:deploy')
+  .addParam('name', 'name of the contract to be deployed', '', paramTypes.string, true)
+  .addParam(
+    'chain',
+    'ckb rpc url or name [default: dev]  [possible values: dev, test, mainnet, http://localhost:8114 etc...]',
```

Comment:
Better to name them clearly. `test` is for `testnet` but may be misleading for others to regard it as a local test environment